### PR TITLE
Limit to FASTMath-only packages

### PR DESCRIPTION
**Pull request by markcmiller86**
As requested per @asalmgren

### CODE REVIEW

File: var/spack/repos/builtin/packages/fastmath/package.py [hunk 1]
```diff
@@ -19,13 +19,11 @@ class Fastmath(Package):
     version('1.0', 'e2b724dfcc31d735897971db91be89ff')
 
     # BundlePackage
-    depends_on('boxlib dims=3')
+    depends_on('amrex')  # default is 3 dimensions
```
Comment (balay) [inline var/spack/repos/builtin/packages/fastmath/package.py]: I'm not sure how robust you want fastmath package to be.
for ex: amrex has a conflict with clang. And its not easy to encode this in the metapackage.
xsdk enables amrex only for gcc and intel compiler.

Comment (markcmiller86) [reply var/spack/repos/builtin/packages/fastmath/package.py]: Sorry, I resolved this and it didn't take my comment. Not looking for any more robustness than existed prior to this change. I think if it works only on vanilla gnu, that is all that is expected.

**Comment by markcmiller86** [comment]
@tgamblin ... this is teeny/tiny and the codecov check I think is bogus -- the result of removal of some lies. I think this is good to go.

**Comment by balay** [comment]
@markcmiller86 I've merged this PR [but modified the commit message a bit]

**Comment by tzanio** [comment]
@markcmiller86, can you please add mfem to the fastmath package

**Comment by markcmiller86** [comment]
@tzanio ... possibly. Don't wanna sound uncollaborative but still strategizing how best to support fastmath and need to wait until some questions are answered.

**Pull request merged**